### ISSUE TITLE
Question about data in validation and test set

### ISSUE THREAD

**Issue by Zhiwei-Zhai** [YingChen7/SELMA3D_challenge-submission-example#2]
Hi Ying,

Thank you for sharing the submission example. In the example, the input data is '.mha' format and output is saved to '.mha' as well. However, the '.nii.gz' format were used for both second training set and validata set. Which format will be used in the test set?

In the discription of challenge Dataset, it mentioned that there will be cells labeled by neural activity marker and microglia marker.  In the example, the input folder structure is 'test/input/images/3d-brain-microscopy'. But it is 'validation/microglia/raw/' for validation set. Could you give a bit more information about the folder structure for validation and test set?

Best regards,
Zhiwei

**Comment by YingChen7** [comment]
Hi Zhiwei,

Yes, the patches we shared earlier for the training and validation sets are in '.nii.gz' format, but there's no need to worry. The Grand Challenge will internally convert all images to .mha format. As shown in the inference.py example, the load_image_file_as_array function will automatically load the validation images once you submit your algorithm container. You don't need to change the image reading and saving parts in inference.py.

The folder structure 'test/input/images/3d-brain-microscopy' is used for testing (via test_run.sh) whether your algorithm works. The validation set we provided earlier is to give participants an idea of how the validation patches look, so 'validation/microglia/raw/' is not the final folder structure. However, the folder structure here is not important. To adopt this example to your algorithm, what you need to do is to modify the following part in the inference.py file, replacing it with your algorithm to make a prediction for the loaded image array:
  ```
  # For now, let us set make bogus predictions
    biological_brain_structure = numpy.eye(4, 2)
```
Then run the test_run.sh for a sanity check. After that you can run the save.sh to create a container of your algorithm and submit the container to the grand challenge. 

Best regards,
Ying Chen

**Comment by YunhengWu-IB** [comment]
Hello Ying,

I hope this message finds you well. I am writing to report a couple of issues I encountered during my Docker submission on Grand Challenge. I received a failure message indicating, "The algorithm failed on one or more cases." 

Therefore, 

1. I would like to confirm whether all the test data consists of 3D images. 

2. Should I save my results in the range **[0, 1] or [0, 255]?**

I would greatly appreciate any insights or confirmation on these points. Thank you in advance for your help!

Best regards, Yunheng

I would greatly appreciate you could provide regarding these issues. Thank you in advance for your help.

Best regards,
Yunheng

**Comment by YingChen7** [comment]
Hi Yunheng,

The test data consists of 3D images, and the output results should be binary, meaning they should be within the range [0, 1].

Best,
Ying
